pylspclient/lsp_client.py

```diff
@@ -15,7 +15,16 @@ def __init__(self, lsp_endpoint: LspEndpoint):
         self.lsp_endpoint = lsp_endpoint
 
 
-    def initialize(self, processId, rootPath, rootUri, initializationOptions, capabilities, trace, workspaceFolders):
+    def initialize(
+        self,
+        processId: Optional[int] = None,
+        rootPath: Optional[str] = None,
+        rootUri: Optional[str] = None,
+        initializationOptions: Optional[Any] = None,
+        capabilities: Optional[dict] = None,
+        trace: Optional[str] = None,
+        workspaceFolders: Optional[list] = None,
+    ):
```

Comment:
We would not need the runtime check if capabilities was reordered to top as required.